src/sentry/migrations/0073_migrate_alert_query_model.py

```diff
@@ -0,0 +1,77 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.29 on 2020-05-06 21:19
+from __future__ import unicode_literals
+
+from datetime import timedelta
+
+from django.db import migrations, transaction
+from enum import Enum
+
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+class QueryAggregations(Enum):
+    TOTAL = 0
+    UNIQUE_USERS = 1
+
+
+aggregation_function_translations = {
+    QueryAggregations.TOTAL: "count()",
+    QueryAggregations.UNIQUE_USERS: "count_unique(user)",
+}
+
+
+def convert_alert_rule_to_snuba_query(alert_rule, SnubaQuery):
+    if alert_rule.snuba_query:
+        return
+
+    with transaction.atomic():
+        try:
+            environment = alert_rule.environment.all()[0]
+        except IndexError:
+            environment = None
+        snuba_query = SnubaQuery.objects.create(
+            dataset=alert_rule.dataset,
+            query=alert_rule.query,
+            aggregate=aggregation_function_translations[QueryAggregations(alert_rule.aggregation)],
+            time_window=int(timedelta(minutes=alert_rule.time_window).total_seconds()),
+            resolution=int(timedelta(minutes=alert_rule.resolution).total_seconds()),
+            environment=environment,
+        )
+        alert_rule.snuba_query = snuba_query
+        alert_rule.save()
+        alert_rule.query_subscriptions.all().update(snuba_query=snuba_query)
+
+
+def migrate_alert_query_model(apps, schema_editor):
+    AlertRule = apps.get_model("sentry", "AlertRule")
+    SnubaQuery = apps.get_model("sentry", "SnubaQuery")
+    for alert_rule in RangeQuerySetWrapperWithProgressBar(
+        AlertRule.objects_with_snapshots.filter(snuba_query__isnull=True)
```

Comment:
We use `objects_with_snapshots` here since it's the default manager. We want to update snapshots as well.

Reply:
Would it be more efficient to include `select_related("snuba_query", "environment")` here?

Reply:
Yeah, I'm not too concerned about performance here since there's a tiny number of rows, but may as well add this

Reply:
Dang, forgot to add this, but it won't be a big deal for <100 rows.